### PR TITLE
adds gradient accumulation to distillation training via tunix

### PR DESCRIPTION
**Pull request by entrpn**
# Description

FIXES [b/490478748](https://buganizer.corp.google.com/issues/490478748) which enables gradient accumulation to the distillation training script.

# Tests

Ran `distillation_checkpointing_test` and `train_distill_test`. 

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/maxtext/trainers/post_train/distillation/train_distill.py [hunk 1]
```diff
@@ -486,6 +491,7 @@ def labels_fn(targets, targets_segmentation=None, **kwargs):
       profiler_options=profiler_options,
       checkpoint_root_directory=student_config.checkpoint_dir,
       checkpointing_options=checkpointing_options,
+      gradient_accumulation_steps=global_config.gradient_accumulation_steps,
```
Comment (vlad-karp) [inline src/maxtext/trainers/post_train/distillation/train_distill.py]: can gradient_accumulation_steps be a part of student_config instead of adding a new global one?

File: src/maxtext/trainers/post_train/distillation/train_distill.py [hunk 2]
```diff
@@ -204,6 +204,7 @@ def __init__(self, model, strategy, optimizer, training_config, **kwargs):
     self.strategy = strategy
 
     # override optimizer to only use student_model.
+    optimizer = optax.MultiSteps(optimizer, training_config.gradient_accumulation_steps)
```
Comment (vlad-karp) [inline src/maxtext/trainers/post_train/distillation/train_distill.py]: may be init MultiSteps optimizer only when gradient_accumulation_steps>1 ?

File: src/maxtext/trainers/post_train/distillation/train_distill.py (hunk 1 above)
Comment (entrpn) [reply src/maxtext/trainers/post_train/distillation/train_distill.py]: done

File: src/maxtext/trainers/post_train/distillation/train_distill.py (hunk 2 above)
Comment (entrpn) [reply src/maxtext/trainers/post_train/distillation/train_distill.py]: done

**Review by richjames0 — APPROVED** [review]
lgtm

File: src/maxtext/trainers/post_train/distillation/train_distill.py
Comment (gagika) [inline src/maxtext/trainers/post_train/distillation/train_distill.py]: should this also be updated if gradient_accumulation_steps > 1?

total_opt_steps = student_config.steps // student_config.gradient_accumulation_steps
optimizer = get_distillation_optimizer(student_config, total_opt_steps)
